velox/type/DecimalUtil.h

```diff
@@ -84,6 +84,35 @@ class DecimalUtil {
     }
   }
 
+  template <typename TOutput>
+  inline static std::optional<TOutput> rescaleBigint(
+      const int128_t inputValue,
+      const int toPrecision,
+      const int toScale,
+      const bool nullOnFailure) {
+    int128_t rescaledValue = inputValue;
```

Comment:
Add static_assert using is_same_v to make sure this method only applies to UnscaledShort and UnscaledLong types. Currently the TOutput can be anything.

Reply:
Thanks for the feedback! This is resolved.